core/common/src/main/java/alluxio/conf/RemovedKey.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import alluxio.grpc.Scope;
+
+/**
+ * This class contains {@link PropertyKey} which should all be annotation with {@link Removed}.
+ *
+ * This class is used to track keys which were deprecated in previous versions, and subsequently
+ * removed in a future version. We still keep them here so that it is possible to provide users
+ * with useful information if they are found to be using an outdated property key.
+ */
+public class RemovedKey {
+
+  private static final String V2_REMOVED = "v2.0 removed the ability to configure this parameter.";
+
+  @Removed(message = "The alluxio web UI overhaul in v2.0 removed this parameter.")
+  public static final PropertyKey WEB_TEMP_PATH =
```

Comment:
why do we keep the whole PropertyKey definition? It seems like we should only need the name/aliases

Reply:
on that note, annotations seem like overkill since we could have a hardcoded `Map<keyName: String, message: String>`. What do annotations get us?

Reply:
See new updates in  331d111